crates/prover/src/task/mod.rs

```diff
@@ -16,6 +19,8 @@ pub trait ProvingTask: serde::de::DeserializeOwned {
     fn identifier(&self) -> String;
 
     fn build_guest_input(&self) -> Result<StdIn, rkyv::rancor::Error>;
+
+    fn fork_name(&self) -> ForkName;
```

Comment:
@noel2004 every impl of `fork_name` is the same. I will suggest just add it here:

```suggestion
    fn fork_name(&self) -> ForkName {
        ForkName::from(self.fork_name.to_lowercase().as_str())
    }
```

Reply:
Then the changes in `task/chunk.rs`, `task/batch.rs` and `task/bundle.rs` should be gone.